cityofaustin/atd-moped
# Add project development status and date fields to AGOL database view

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/17605

This adds two fields to `component_arcgis_online_view` called `project_development_status` and `project_development_status_date`. They are used to capture the next upcoming stage of project development as well as an estimated status date for ATSD reporting purposes.

## Testing
**URL to test:** 

Local

**Steps to test:**
1. Create a new project and add one component
2. Add a **Complete** phase to your project and mark it as the current phase and make sure it has a confirmed start date
3. Run the query below and you should see:
   - `project_development_status` is "Complete"
   - `project_development_status_date` is the start date of your **Complete** phase
4. Now uncheck the **Confirmed** box next to the start date of your **Complete** phase to make it estimated. Make a new **Construction** phase and mark it as the current phase.
5. Run the query below and you should see:
   - `project_development_status` is "Estimated End Date (In Construction)"
   - `project_development_status_date` is the estimated date in your **Complete** phase
6. Update your **Construction** phase to **Planned** (or any phase that does not have a `phase_name_simple` of Complete or Construction in the `moped_phases` table)
7. Run the query below and you should see:
   - `project_development_status` is "Estimated End Date"
   - `project_development_status_date` is still the estimated date in your **Complete** phase

8. Now remove all phases, and then add a milestone that is **Public Meeting** with an estimated completion date.
9. Run the query below and you should see:
   - `project_development_status` is "Estimated Public Meeting Date"
   - `project_development_status_date` is your estimated date
10. Add a **Date completed** date to this same milestone and re-run the query, and you should see the `project_development_status_date` update with the completed date.
11. Delete your milestone and add a phase that is **Construction** or **Scoping** (or any phase that has `phase_name_simple` of Active or Construction) and make it the current phase
12. Run the query below and you should see:
   - `project_development_status` is "Estimated Start of Project Development"
   - `project_development_status_date` is the date that you added to your phase (estimated or confirmed)
13. Test the down migration


```sql
SELECT
	project_id,
	project_component_id,
	project_development_status,
	project_development_status_date
FROM
	component_arcgis_online_view
WHERE
	project_id = <your project id>;
```

#### Ship list
